File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    answer = 0
    l = len(elements)
    elements *= 2
    ss = set()

    
    for i in range(1,l+1):
        for s in range(l):
            ss.add(sum(elements[s:s+i]))
    
    
    return len(ss)

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.swift
import Foundation

func solution(_ elements:[Int]) -> Int {
    
    let k = elements.count
    var answer: Int = 0
    var elements = elements + elements
    

    var set = Set<Int>()
    
    for start in 0..<k {
        
        var sum : Int = 0 
        for l in 0..<k {
            sum += elements[start+l]
            set.insert(sum)
        }
        
    }
    
    return set.count
}